File: src/topic02_tf_tutorials/transform_publisher/frame_a_to_frame_b_listener.cpp
#include <ros/ros.h>
#include <tf/transform_listener.h>
#include <tf/tf.h>

int main (int argc, char** argv)
{
    ros::init(argc, argv, "listener");
    ros::NodeHandle nh;

    tf::TransformListener listener;
    tf::StampedTransform transformation;

    ros::Rate rate(1);

    tf::Vector3 origin;
    tf::Quaternion quat;

    double roll;
    double pitch;
    double yaw;

    while(ros::ok()) {
        try {
            listener.lookupTransform("frame_a", "frame_b", ros::Time(0), transformation);
            origin = transformation.getOrigin();
            quat = transformation.getRotation();
            // printf("\nX: %f, Y: %f", origin.x(), origin.y());
            // printf("\nX: %f, W: %f", quat.x(), quat.w());
            tf::Matrix3x3 m(quat);
            m.getRPY(roll, pitch, yaw);
            printf("\nAngle: [%f, %f, %f]", roll, pitch, yaw);
        }
        catch (tf::TransformException ex) {
            ROS_ERROR("%s",ex.what());
            // ros::Duration(1.0).sleep();
        }
        rate.sleep();
    }
    return 0;
}

File: src/topic02_tf_tutorials/transform_publisher/frame_a_to_frame_b_listener.py
#!/usr/bin/env python  
import rospy
import math
import tf
import geometry_msgs.msg
import turtlesim.srv

if __name__ == '__main__':
    rospy.init_node('frame_a_frame_b_listener_node')

    listener = tf.TransformListener()
    rate = rospy.Rate(1.0)
    listener.waitForTransform('/frame_a', '/frame_b', rospy.Time(), rospy.Duration(4.0))
    
    while not rospy.is_shutdown():
        try:
            (trans,rot) = listener.lookupTransform('/frame_a', '/frame_b', rospy.Time(0))
        except (tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException):
            continue

        rpy=tf.transformations.euler_from_quaternion(rot)
        print("\ntranslation: %f, %f, %f"%(trans[0],trans[1],trans[2]))
        print(type(trans))
        # print 'rotation angles: roll=',rpy[0],' pitch=',rpy[1],' yaw=',rpy[2]


        rate.sleep()